larq_compute_engine/mlir/transforms/optimize.cc

```diff
@@ -34,34 +39,152 @@ bool IsConstantValue(Attribute values, float expected_value) {
   return true;
 }
 
-DenseElementsAttr ComputeBSignAndExpandTo4D(Attribute attr) {
-  auto tensor = attr.cast<DenseElementsAttr>();
-  auto channels = tensor.getNumElements();
-  auto tensor_type = attr.getType().cast<ShapedType>();
+#include "larq_compute_engine/mlir/transforms/generated_optimize.inc"
 
-  std::vector<APFloat> results;
-  results.reserve(channels);
-  for (auto value : tensor.getValues<float>()) {
-    auto result = std::signbit(value) ? -1.0f : 1.0f;
-    results.push_back(APFloat(result));
-  }
+/**
+ * =================================================
+ * Computing thresholds for writing bitpacked output
+ * =================================================
+ *
+ * Consider a single output element of a binary convolution, y. We have that:
+ *
+ *                       y = clamp(a - 2x, c, C) * μ + β
```

Comment:
This is a super nit-picky suggestion, but do you mind changing mu to gamma to make it consistent with the formulas in the original batch norm paper? I can't help it but mu, always makes me think of a mean instead of a multiplier.

Reply:
Sure, happy to :+1: